DSCResources/MSFT_xDisk/MSFT_xDisk.psm1

```diff
@@ -144,8 +144,16 @@ function Set-TargetResource
         else 
         {
             Write-Verbose -Message "The volume already exists, adjusting drive letter..."
+            if (($Disk | Get-Partition | Get-Volume).driveletter -ne $null) 
```

Comment:
Instead of getting this value twice (once for checking and once for assignment), can we not do it just once? That also makes the if condition simpler to read